### DISCUSSION TITLE
Web guard + API guard for browser extension

### DISCUSSION THREAD

**Discussion by Dolu89** [adonisjs/core#1560] [Help] [answered]
Hi
I want to have the web guard for my website to manage my users (using sessions) and the API guard to manage a public API inside the same project (used by a browser extension I will create later)

There is no problem on the website (register and login on webpage are working), but when I'm trying to call my API without bearer token (in postman for example), I got a redirection to login route

![image](https://user-images.githubusercontent.com/4184590/92301605-60069180-ef65-11ea-8f77-390c7fd16ca6.png)

Here is the route.ts
``` ts
Route.group(() => {

    Route.group(() => {
        Route.post('/video', 'ApiV1Controller.video')
    }).prefix('youtube')

}).prefix('api/v1').middleware('auth:api')

```

I would have thought that using `middleware('auth:api')`, the response would have been a JSON result
Did I miss something?
Thanks

**Comment by thetutlage** [comment, marked as answer]
You need to use content negotiation for that. Send `Accept: application/json` header in your request

**Reply by Dolu89** [reply]
Thank you so much!
